spec/support/jazz.rb

```diff
@@ -304,6 +312,11 @@ def self.resolve_type(object, context)
 
   class HashKeyTest < BaseObject
     field :falsey, Boolean, null: false
+
+    def falsey
+      # test that super can be used in objects
+      super
+    end
```

Comment:
I would suggest a couple test cases, mostly as sanity checks in case someone alters this code:

1. If I set up two chains of interfaces (A, B, and D are interfaces):
- `C (class object type) < B < A < GraphQL::Schema::Interface`
- `E (class object type) < D < A < GraphQL::Schema::Interface`
- `F (class object type) < GraphQL::Schema::Object`

Then I should be able to:
- Have a field on `A` that is callable from both `C` and `E`, but not `F`
- Have a field on `B` that is callable from `C` but not `E` or `F (interfaces don't cross-pollute)
- Have a field on `C` that is callable from `C`, but not `E` or `F` (objects don't cross-pollute)

2. Same thing, but make A, B, and D superclasses instead of interfaces.

Reply:
I think that would make sure that all of the book-keeping is set up, and also future-proof it in case someone refactors it.

Reply:
Thanks for taking a look and making this great suggestion, I made some proper tests in 7674a590. I think the specific inheritance chains were a bit different, but I think I covered all the bases!